File: app-ios/VelaWallet/VelaWallet/Components/Wallet/ActionButtonRow.swift
//
//  ActionButtonRow.swift
//  VelaWallet
//
//  ActionButtonRow (spec 015 vocabulary #6): 收款 / 转账 / 扫码 as three
//  equal raised cards, icon above label. Taps are inert in this feature
//  (no destination mocks on mobile — spec assumption).
//

import SwiftUI

/// One card of the row: glyph above label. Spec 018 reuses the component
/// with its own items (转账 / 收款 / 二维码) instead of a second row type.
struct ActionCardItem: Identifiable {
    let id = UUID()
    let icon: LucideGlyph
    let label: String
    /// Spec 021: the card is an entry into a flow. Absent in the gallery,
    /// where the dock is a picture.
    var action: (() -> Void)?
}

struct ActionButtonRow: View {
    @Environment(\.theme) private var theme
    @Environment(\.walletTextScale) private var textScale

    let items: [ActionCardItem]

    init(items: [ActionCardItem]) {
        self.items = items
    }

    /// Wallet home (spec 015): 收款 / 转账 / 扫码.
    init(
        model: ActionsModel,
        onReceive: (() -> Void)? = nil,
        onSend: (() -> Void)? = nil,
        onScan: (() -> Void)? = nil
    ) {
        self.items = [
            ActionCardItem(icon: .arrowDownLeft, label: model.receive, action: onReceive),
            ActionCardItem(icon: .arrowUpRight, label: model.send, action: onSend),
            ActionCardItem(icon: .scanLine, label: model.scan, action: onScan),
        ]
    }

    var body: some View {
        HStack(spacing: Tokens.Space.s12) {
            ForEach(items) { item in
                if let action = item.action {
                    Button(action: action) {
                        card(icon: item.icon, label: item.label).contentShape(Rectangle())
                    }
                    .buttonStyle(.plain)
                } else {
                    card(icon: item.icon, label: item.label)
                }
            }
        }
    }

    private func card(icon: LucideGlyph, label: String) -> some View {
        VStack(spacing: Tokens.Space.s8) {
            LucideIcon(icon, size: LucideIconSize.action)
                .foregroundStyle(theme.fgBase)
            Text(verbatim: label)
                .typeRole(Typography.actionLabel.scaled(textScale))
                .foregroundStyle(theme.fgBase)
                .lineLimit(1)
        }
        .frame(maxWidth: .infinity)
        .padding(.vertical, Tokens.Space.s12)
        .frame(minHeight: WalletGeometry.actionCardHeight)
        .background(RoundedRectangle(cornerRadius: Tokens.Radius.r16).fill(theme.bgRaised))
    }
}

#Preview("Actions dark") {
    ActionButtonRow(model: ActionsModel(receive: "收款", send: "转账", scan: "扫码"))
        .padding(Tokens.Space.s24)
        .background(Tokens.dark.bgBase.color)
        .themed(.dark)
}

#Preview("Actions light") {
    ActionButtonRow(model: ActionsModel(receive: "Receive", send: "Send", scan: "Scan"))
        .padding(Tokens.Space.s24)
        .themed(.light)
}

File: app-android/vela-wallet/app/src/main/java/app/getvela/wallet/feature/wallet/components/ActionButtonRow.kt
package app.getvela.wallet.feature.wallet.components

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import app.getvela.wallet.core.designsystem.components.VelaCard
import app.getvela.wallet.core.designsystem.components.VelaIcons
import app.getvela.wallet.core.designsystem.theme.VelaTheme
import app.getvela.wallet.core.designsystem.tokens.VelaFontFamily
import app.getvela.wallet.core.designsystem.tokens.VelaFontWeight
import app.getvela.wallet.core.designsystem.tokens.VelaIconSize
import app.getvela.wallet.core.designsystem.tokens.VelaSpacing
import app.getvela.wallet.core.designsystem.tokens.VelaTextSize
import app.getvela.wallet.feature.wallet.ActionsModel

/** One equal-width action card: glyph above label. */
data class ActionButtonItem(
    val icon: ImageVector,
    val label: String,
    val onClick: () -> Unit = {},
)

/**
 * 收款 / 转账 / 扫码 dock (spec vocabulary #6): three equal raised cards,
 * icon above label. Taps have no destination yet (spec assumption).
 */
@Composable
fun ActionButtonRow(
    actions: ActionsModel,
    modifier: Modifier = Modifier,
    onReceive: () -> Unit = {},
    onSend: () -> Unit = {},
    onScan: () -> Unit = {},
) {
    ActionButtonRow(
        items = listOf(
            ActionButtonItem(VelaIcons.ArrowDownLeft, actions.receive, onReceive),
            ActionButtonItem(VelaIcons.ArrowUpRight, actions.send, onSend),
            ActionButtonItem(VelaIcons.ScanLine, actions.scan, onScan),
        ),
        modifier = modifier,
    )
}

/**
 * Item-driven form of the same dock — spec 018 reuses it for the contact
 * detail's 转账 / 收款 / 二维码 cards (spec vocabulary #11 "[reuse] the 015
 * component with new items"), so there is still one card implementation.
 */
@Composable
fun ActionButtonRow(
    items: List<ActionButtonItem>,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier = modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.spacedBy(VelaSpacing.lg),
    ) {
        items.forEach { item ->
            ActionCard(
                icon = item.icon,
                label = item.label,
                onClick = item.onClick,
                modifier = Modifier.weight(1f),
            )
        }
    }
}

@Composable
private fun ActionCard(
    icon: ImageVector,
    label: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    val colors = VelaTheme.colors
    VelaCard(modifier = modifier) {
        Column(
            modifier = Modifier
                .clickable(onClick = onClick)
                .fillMaxWidth()
                .padding(vertical = VelaSpacing.xl),
            horizontalAlignment = Alignment.CenterHorizontally,
        ) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                tint = colors.fgBase,
                modifier = Modifier.size(VelaIconSize.lg),
            )
            Spacer(modifier = Modifier.height(VelaSpacing.md))
            Text(
                text = label,
                color = colors.fgBase,
                fontFamily = VelaFontFamily,
                fontWeight = VelaFontWeight.medium,
                fontSize = VelaTextSize.base,
                maxLines = 1,
            )
        }
    }
}
